src/gui/attributetable/qgsfeaturelistmodel.cpp

```diff
@@ -61,10 +61,14 @@ QVariant QgsFeatureListModel::data( const QModelIndex &index, int role ) const
     {
       return QSettings().value( "qgis/nullValue", "NULL" ).toString();
     }
-    else
+    else if ( role == QgsAttributeTableModel::FeatureIdRole )
```

Comment:
Unfortunately not, because the different roles expect different types and whatever type we would give the _null QVariant_, it would be wrong for most roles.
Examples I found through try and error:
* `Qt::CheckStateRole` interprets a `QVariant( QVariant::Int )` in a way that makes it draw a checkbox
* `Qt::SizeHintRole` interprets a `QVariant( QVariant::Int )` in a way that hides the entry